lib/TPP/ConvertTppToLoops.cpp

```diff
@@ -22,49 +22,75 @@ using namespace mlir::tpp;
 
 namespace {
 
+static bool isScalarVal(Value val) { return !val.getType().isa<ShapedType>(); }
```

Comment:
is there a similar one elsewhere? candidate for utils?

Reply:
Looks like we use this only when converting from tpp to loops.